Flick/Controllers/DiscoverSearchViewController.swift

```diff
@@ -0,0 +1,176 @@
+        tabCollectionView.showsHorizontalScrollIndicator = false
+        view.addSubview(tabCollectionView)
+
+        let pageCollectionViewLayout = UICollectionViewFlowLayout()
+        pageCollectionViewLayout.scrollDirection = .horizontal
+
+        searchResultPageCollectionView = UICollectionView(frame: .zero, collectionViewLayout: pageCollectionViewLayout)
+        searchResultPageCollectionView.backgroundColor = .offWhite
+        searchResultPageCollectionView.register(UICollectionViewCell.self, forCellWithReuseIdentifier: searchResultPageReuseIdentifier)
+        searchResultPageCollectionView.dataSource = self
+        searchResultPageCollectionView.delegate = self
+        searchResultPageCollectionView.showsHorizontalScrollIndicator = false
+        searchResultPageCollectionView.isPagingEnabled = true
+        view.addSubview(searchResultPageCollectionView)
+
+        setupConstraints()
+        setupViewControllers()
+    }
+
+    private func setupNavigationBar() {
+        let backButtonSize = CGSize(width: 22, height: 18)
+
+        navigationController?.setNavigationBarHidden(false, animated: true)
+        navigationController?.navigationBar.barTintColor = .offWhite
+        navigationController?.navigationBar.shadowImage = UIImage()
+
+        let backButton = UIButton()
+        backButton.setImage(UIImage(named: "backArrow"), for: .normal)
+        backButton.tintColor = .black
+        backButton.snp.makeConstraints { make in
+            make.size.equalTo(backButtonSize)
+        }
+
+        backButton.addTarget(self, action: #selector(backButtonPressed), for: .touchUpInside)
+        let backBarButtonItem = UIBarButtonItem(customView: backButton)
+        navigationItem.leftBarButtonItem = backBarButtonItem
+
+        searchBar.delegate = self
+        navigationItem.titleView = searchBar
+    }
+
+    private func setupConstraints() {
+        tabCollectionView.snp.makeConstraints { make in
+            make.top.equalTo(view.safeAreaLayoutGuide.snp.top)
+            make.leading.trailing.equalToSuperview().inset(10)
+            make.height.equalTo(40)
+        }
+
+        searchResultPageCollectionView.snp.makeConstraints { make in
+            make.top.equalTo(tabCollectionView.snp.bottom).offset(15)
+            make.leading.trailing.equalToSuperview()
+            make.bottom.equalTo(view.safeAreaLayoutGuide.snp.bottom)
+        }
+    }
+
+    func setupViewControllers() {
+        tabs.forEach { tab in
+            searchResultViewControllers.append(DiscoverSearchResultViewController(seachTab: tab))
+        }
+    }
+
+    func setCurrentPosition(position: Int){
+        currentPosition = position
+        let path = IndexPath(item: currentPosition, section: 0)
+
+        DispatchQueue.main.async {
+            self.tabCollectionView.reloadData()
+        }
+
+        DispatchQueue.main.async {
+            self.searchResultPageCollectionView.scrollToItem(at: path, at: .centeredHorizontally, animated: true)
```

Comment:
What is the reason that we need async?

Reply:
hmm i'm not entirely sure, it might be so the scrolling doesn't interfere with other things that goes on in the view?? 
i just followed this medium post lol https://medium.com/@erthru/implement-android-like-sliding-tabs-in-swift-xcode-13061f5444fc